dsl/src/main/scala/org/http4s/dsl/Path.scala

```diff
@@ -93,7 +93,7 @@ object ~ {
 final case class /(parent: Path, child: String) extends Path {
   lazy val toList: List[String] = parent.toList ++ List(child)
   def lastOption: Option[String] = Some(child)
-  lazy val asString = parent.toString + "/" + UrlCodingUtils.urlEncode(child, toSkip = Path.pathUnreserved)
+  lazy val asString = s"${parent}/${Uri.pathEncode(child, Charset.`UTF-8`)}"
```

Comment:
Maybe this case class needs a charset?  Would have to be an implicit parameter to the constructor, I think.

It was hardcoded UTF-8 before, and nobody complained.